File: client/src/utils/telecrmSmsPlaceholders.ts
export type TelecrmSmsPlaceholderCtx = {
  customerName?: string;
  phone?: string;
  pyeong?: string;
  estimate?: string;
  orderLink?: string | null;
};

/** `{고객명}` `{연락처}` `{평수}` `{예상가}` `{발주서링크}` 치환 */
export function applyTelecrmSmsPlaceholders(body: string, ctx: TelecrmSmsPlaceholderCtx): string {
  return body
    .replace(/\{고객명\}/g, ctx.customerName?.trim() || '고객님')
    .replace(/\{연락처\}/g, ctx.phone?.trim() || '—')
    .replace(/\{평수\}/g, ctx.pyeong?.trim() || '—')
    .replace(/\{예상가\}/g, ctx.estimate?.trim() || '—')
    .replace(/\{발주서링크\}/g, ctx.orderLink?.trim() || '(발주서 링크 없음)');
}

File: apps/telecrm-android/app/src/main/java/com/skcleantec/telecrm/ui/TelecrmSmsPlaceholders.kt
package com.skcleantec.telecrm.ui

data class TelecrmSmsPlaceholderCtx(
    val customerName: String? = null,
    val phone: String = "",
    val pyeong: String? = null,
    val estimate: String? = null,
    val orderLink: String? = null,
)

object TelecrmSmsPlaceholders {
    fun apply(body: String, ctx: TelecrmSmsPlaceholderCtx): String =
        body
            .replace("{고객명}", ctx.customerName?.trim()?.takeIf { it.isNotBlank() } ?: "고객님")
            .replace("{연락처}", ctx.phone.trim().takeIf { it.isNotBlank() } ?: "—")
            .replace("{평수}", ctx.pyeong?.trim()?.takeIf { it.isNotBlank() } ?: "—")
            .replace("{예상가}", ctx.estimate?.trim()?.takeIf { it.isNotBlank() } ?: "—")
            .replace("{발주서링크}", ctx.orderLink?.trim()?.takeIf { it.isNotBlank() } ?: "(발주서 링크 없음)")
}
